src/structures/paging/recursive.rs

```diff
@@ -51,9 +62,37 @@ pub trait Mapper<S: PageSize> {
 
     /// Removes a mapping from the page table and returns the frame that used to be mapped.
     ///
-    /// Note that no page tables or pages are deallocated.
+    /// Note that no page tables or other frames are deallocated.
     fn unmap(&mut self, page: Page<S>) -> Result<(PhysFrame<S>, MapperFlush<S>), UnmapError>;
 
+    /// Reclaim page tables for virtual addresses corresponding to the given range of pages.
+    /// The page tables are returned to the given `deallocator`. We reclaim page tables according
+    /// to `mode`, which describes what behvior to take when we encounter a page table entirely in
+    /// the given range with a used (i.e. non-zero) page table entry. See the docs for
+    /// `ReclaimPageTablesMode` for more info on modes.
+    ///
+    /// Note that it is the caller's responsibility to make sure the page tables are not in use.
+    /// Don't forget to check shared mappings!
+    ///
+    /// It's also worth noting that the p4 table currently in use can never be reclaimed by this
+    /// function, since it is being used. Thus, if you need to reclaim a p4 table, you must do so
+    /// from another address space.
+    ///
+    /// We only try to reclaim page tables whose mappings would all lie entirely within the range.
+    ///
+    /// # Panics
+    ///
+    /// - If one of the range enpoints is `Unbounded`.
+    /// - If the `mode` specifies that we should.
+    unsafe fn reclaim_page_tables<D, R>(
```

Comment:
I'm not sure whether this should be a method of the `Mapper` trait. It does not really have something to do with mapping/unmapping and it is completely independent of the generic `PageSize` parameter. Simply adding it as a method to `RecursivePageTable` might be a better fit and would also avoid the identical implementations for all page sizes.